Code/GraphMol/RWMol.cpp

```diff
@@ -281,12 +283,54 @@ void RWMol::removeAtom(Atom *atom) {
   }
   // now deal with bonds:
   //   their end indices may need to be decremented and their
-  //   indices will need to be handled
+  //   indices will need to be handled and if they have an
+  //   ENDPTS prop that includes idx, it will need updating.
   unsigned int nBonds = 0;
   EDGE_ITER beg, end;
   boost::tie(beg, end) = getEdges();
+  std::string sprop;
   while (beg != end) {
     Bond *bond = d_graph[*beg++];
+    if (bond->getPropIfPresent(RDKit::common_properties::_MolFileBondEndPts,
+                               sprop)) {
+      // This would ideally use ParseV3000Array but I'm buggered if I can get
+      // the linker to find it.
+      //      std::vector<unsigned int> oats =
+      //          RDKit::SGroupParsing::ParseV3000Array<unsigned int>(sprop);
+      if ('(' == sprop.front() && ')' == sprop.back()) {
+        sprop = sprop.substr(1, sprop.length() - 2);
```

Comment:
I made a suggestion about using `string_view` in MolOps.cpp, you can do that here too